src/Aurora.Infrastructure/Scrapers/PornhubScraper.cs

```diff
@@ -0,0 +1,273 @@
+                var searchTermUrlFormatted = FormatTermToUrl(searchTerm);
+                var searchPageUrl = $"{_baseUrl}/video/search?search={searchTermUrlFormatted}&page={pageNumber}";
+                await _clientProvider.SetUserString(client);
+                var htmlSearchPage = client.DownloadString(searchPageUrl);
+                htmlDocument.LoadHtml(htmlSearchPage);
+
+                var videoLinksNodes = htmlDocument.DocumentNode
+                    ?.SelectSingleNode("//body")
+                    ?.SelectNodes("//a[contains(@class, 'videoPreviewBg')]");
+
+                if (videoLinksNodes is null)
+                {
+                    break;
+                }
+
+                foreach (var videoLinkNode in videoLinksNodes)
+                {
+                    SearchItem searchVideoItem = new()
+                    {
+                        Option = SearchOption.Video
+                    };
+
+                    var currentLinkImageNode = videoLinkNode.ChildNodes
+                        .FirstOrDefault(n => n.Name == "img");
+
+                    if (currentLinkImageNode != null)
+                    {
+                        var currentLinkImageAttributes = currentLinkImageNode.Attributes;
+                        searchVideoItem.ImagePreviewUrl = currentLinkImageAttributes["data-thumb_url"].Value;
+                    }
+
+                    var currentLinkAttributes = videoLinkNode.Attributes;
+                    searchVideoItem.SearchItemUrl = $"{_baseUrl}{currentLinkAttributes["href"].Value}";
+
+                    urlsCount++;
+                    videoItems.Add(searchVideoItem);
+                }
+
+                pageNumber++;
+                await Task.Delay(1000);
+            }
+
+            return videoItems;
+        }
+
+        private async Task<IEnumerable<SearchItem>> ScrapImages(string requestSearchTerm, int maxNumberOfImageUrls)
+        {
+            var htmlDocument = new HtmlAgilityPack.HtmlDocument
+            {
+                OptionFixNestedTags = true
+            };
+
+            var driver = await _initializer.Initialize();
+            var searchTerm = FormatTermToUrl(requestSearchTerm);
+            var result = new List<SearchItem>();
+
+            using var client = await _clientProvider.Provide();
+            for (var i = 0; i < PAGE_NUMBER_LIMIT; i++)
+            {
+                var pageNumber = i + 1;
+                var fullUrl = GetImagePageUrl(searchTerm, pageNumber);
+                await _clientProvider.SetUserString(client);
+                var html = client.DownloadString(fullUrl);
+                htmlDocument.LoadHtml(html);
+
+                var albumNodes = htmlDocument.DocumentNode
+                    ?.SelectSingleNode("//*[@id='photosAlbumsSection']")
+                    ?.SelectNodes("//li[contains(@class,'photoAlbumListContainer')]/div/a");
+
+                if (albumNodes is null) continue;
```

Comment:
Please use separate block statement for those

Reply:
That`s not something you can just ignore

Reply:
Btw you also wrote like that...